relayer/crates/starknet-integration-tests/src/tests/ics20.rs

```diff
@@ -330,188 +341,54 @@ fn test_starknet_ics20_contract() -> Result<(), Error> {
-                    receive_transfer_event.receiver,
-                    Participant::Native(recipient_address)
-                );
-            }
-
-            let token_address = {
-                let create_token_event = ibc_transfer_events
-                    .iter()
-                    .find_map(|event| {
-                        if let IbcTransferEvent::CreateToken(event) = event {
-                            Some(event)
-                        } else {
-                            None
-                        }
-                    })
-                    .ok_or_else(|| eyre!("expect create token event"))?;
-
-                assert_eq!(create_token_event.initial_supply, amount.into());
-
-                let token_address = create_token_event.address;
-
-                info!("created token address: {:?}", token_address);
-
-                token_address
-            };
-
-            {
-                let recipient_balance = starknet_chain
-                    .query_token_balance(&token_address, &recipient_address)
-                    .await?;
-
-                info!("recipient balance after transfer: {}", recipient_balance);
-
-                assert_eq!(recipient_balance.quantity, amount.into());
-            }
-
-            token_address
-        };
-
-        {
-            // Send the same transfer message a second time
-            // but increase the packet sequence number
-            msg_recv_packet.packet.sequence += 1;
-
-            let calldata = cairo_encoding.encode(&msg_recv_packet)?;
-
-            let call = Call {
-                to: ibc_core_address,
-                selector: selector!("recv_packet"),
-                calldata,
-            };
-
-            let message = StarknetMessage::new(call);
-
-            let response = starknet_chain.send_message(message.clone()).await?;
+        let balance_cosmos_a_step_0 = cosmos_chain
+            .query_balance(address_cosmos_a, denom_cosmos)
+            .await?;
 
-            let ibc_packet_events_2: Vec<PacketRelayEvents> =
-                event_encoding.filter_decode_events(&response.events)?;
+        info!(
+            "cosmos balance before transfer: {:?}",
+            balance_cosmos_a_step_0
+        );
 
-            info!("ibc_packet_events 2: {:?}", ibc_packet_events_2);
+        let packet = <CosmosChain as CanIbcTransferToken<StarknetChain>>::ibc_transfer_token(
+            cosmos_chain,
+            &cosmos_channel_id,
+            &IbcPortId::transfer(),
+            wallet_cosmos_a,
+            address_starknet_b,
+            &Amount::new(transfer_quantity, denom_cosmos.clone()),
+            &None,
+        )
+        .await?;
 
-            let ibc_transfer_events_2: Vec<IbcTransferEvent> =
-                event_encoding.filter_decode_events(&response.events)?;
+        // TODO(rano): how do I get the ics20 token contract address from starknet events
+        cosmos_to_starknet_relay.relay_packet(&packet).await?;
```

Comment:
We can use this:
https://github.com/informalsystems/ibc-starknet/blob/717b22900de39b7efee65f10239da9aee56b248f/cairo-contracts/packages/apps/src/transfer/interfaces/transfer.cairo#L13-L27

Reply:
Ended up with implementing a query endpoint that returns all token addresses, which is just one in the e2e test.